### PR TITLE
ign -> gz Migrate config and log directories : gz-gui

### PR DESCRIPTION
**Pull request by methylDragon**
See: https://github.com/gazebo-tooling/release-tools/issues/757

### CODE REVIEW

File: src/Application.cc [hunk 1]
```diff
@@ -728,10 +728,10 @@ std::vector<std::pair<std::string, std::vector<std::string>>>
   for (auto const &path : this->dataPtr->pluginPaths)
     paths.push_back(path);
 
-  // 3. ~/.ignition/gui/plugins
+  // 3. ~/.gz/gui/plugins
   std::string home;
   common::env(GZ_HOMEDIR, home);
-  paths.push_back(home + "/.ignition/gui/plugins");
+  paths.push_back(home + "/.gz/gui/plugins");
```
Comment (chapulina) [inline src/Application.cc:734]: Here we could tick-tock (i.e. support both `ignition` and `gz` for Garden)

File: src/Application.cc [hunk 2]
```diff
@@ -433,7 +433,7 @@ bool Application::LoadPlugin(const std::string &_filename,
   // Add default folder and install folder
   std::string home;
   common::env(GZ_HOMEDIR, home);
-  systemPaths.AddPluginPaths(home + "/.ignition/gui/plugins:" +
+  systemPaths.AddPluginPaths(home + "/.gz/gui/plugins:" +
```
Comment (chapulina) [inline src/Application.cc:436]: Here too, we can support both directories

Comment (methylDragon) [reply src/Application.cc:464]: https://github.com/gazebosim/gz-gui/pull/454/commits/2d40757457b56309f6cab8c2330ace404481e5ba

File: src/Application.cc (hunk 1 above)
Comment (methylDragon) [reply src/Application.cc:773]: https://github.com/gazebosim/gz-gui/pull/454/commits/2d40757457b56309f6cab8c2330ace404481e5ba

**Comment by nkoenig** [comment]
@methylDragon , did you want to target a different branch?

**Review by nkoenig — APPROVED** [review]

**Comment by chapulina** [comment]
Yeah @methylDragon , let's rebase this onto `gz-sim7`

**Comment by azeey** [comment]
I've rebased this to `gz-gui7` and removed commit 1551520546187bea923e9cd1775c7e3b400a6a52 that belonged in `main` and was causing conflicts.

**Review by azeey — APPROVED** [review]

**Comment by azeey** [comment]
The Github Actions failures also happen in on the `gz-gui7` branch (https://github.com/gazebosim/gz-gui/actions/runs/4389109614/jobs/7686333715) and the hombrew failure is a know issue #503. Merging!

**Pull request merged**